borked_bot/util/twitch.py

```diff
@@ -0,0 +1,67 @@
+from ..credentials import CREDENTIALS
+from .util import *
+from requests.exceptions import ReadTimeout, ConnectionError
+
+MAX_RESULTS = 100
+
+
+class TwitchError(Exception): pass
+
+
+@retry(exceptions=[ConnectionError, ReadTimeout])
+def get_twitch_access_token(session):
+    # https://dev.twitch.tv/docs/authentication/getting-tokens-oauth/#client-credentials-grant-flow
+    params = {
+        "client_id": CREDENTIALS["twitch-client-id"],
+        "client_secret": CREDENTIALS["twitch-secret"],
```

Comment:
are these creds hard to get? do they expire?

Reply:
My PR description explains how to get these. I don't think they expire.

Reply:
FYI, I'm planning on ripping out this code and replacing it with a line or two that initializes `pyTwitchAPI`. The reason being, your YT and Twitter code has basic rate limiting implemented as a certain allowed number of requests per minute. But Twitch has a variable rate limit depending on server capacity, and it's easier to just pull in a library that already implements this. So really, other than your queries, the code for this PR is pretty short. I'm still happy to implement it if it helps you, but if I'm slowing you down, feel free to close this in favor of your own.

Reply:
that's fine.

i'm in no rush to do this. if you don't end up doing it i'll probably eventually get around to it but this isn't the top of my priority list (unless / until the enwiki side of this is ready)